client/core/core.go

```diff
@@ -1860,6 +1846,9 @@ func (c *Core) loadWallet(dbWallet *db.Wallet) (*xcWallet, error) {
 			// of deadlocking a Core method that calls Wallet.Disconnect.
 			go c.tipChange(assetID, err)
 		},
+		PeersChange: func(numPeers uint32) {
+			go c.peerChange(wallet, numPeers)
```

Comment:
@buck54321 I can put a `wallet != nil` check in here.

Reply:
Oh right, because it's not in the `wallets` map until after successful connect and a variety of other operations depending on whos calling `loadWallet`.  :(

Reply:
potentially:

```diff
diff --git a/client/core/core.go b/client/core/core.go
index f81f9895..45ca809d 100644
--- a/client/core/core.go
+++ b/client/core/core.go
@@ -1834,6 +1834,7 @@ func (c *Core) assetDataDirectory(assetID uint32) string {
 // wallet. The returned wallet is running but not connected.
 func (c *Core) loadWallet(dbWallet *db.Wallet) (*xcWallet, error) {
        var wallet *xcWallet
+       var ready uint32
        // Create the client/asset.Wallet.
        assetID := dbWallet.AssetID
        walletCfg := &asset.WalletConfig{
@@ -1847,6 +1848,9 @@ func (c *Core) loadWallet(dbWallet *db.Wallet) (*xcWallet, error) {
                        go c.tipChange(assetID, err)
                },
                PeersChange: func(numPeers uint32) {
+                       if atomic.LoadUint32(&ready) == 0 {
+                               return
+                       }
                        go c.peerChange(wallet, numPeers)
                },
                DataDir: c.assetDataDirectory(assetID),
@@ -1875,6 +1879,7 @@ func (c *Core) loadWallet(dbWallet *db.Wallet) (*xcWallet, error) {
                walletType: dbWallet.Type,
                traits:     asset.DetermineWalletTraits(w),
        }
+       atomic.StoreUint32(&ready, 1)
        return wallet, nil
 }
```

but... :vomiting_face: